src/ogma/features/nodeGrouping.ts

```diff
@@ -38,22 +41,28 @@ export class NodeGroupingTransformation {
           if (this._isRuleNotApplicableToNode(node)) {
             return undefined;
           } else {
-            const propertyValue = node.getData(['properties', this.groupRule?.property ?? '']);
+            this._unpinNode(node);
```

Comment:
is it the correct place to do this? It seems like it's a side effect. In this method you are computing a unique id to group node and should not do something else. Is there a way to do it somewhere else?

Reply:
you are right. It doesn't seem the right place.
I will probably need to do it after the transformation is refreshed